messages/java/src/main/java/io/cucumber/messages/Messages.java

```diff
@@ -0,0 +1,4077 @@
+            this.testStepId = testStepId;
+            this.url = url;
+        }
+
+        public String getBody() {
+            return body;
+        }
+
+        public AttachmentContentEncoding getContentEncoding() {
+            return contentEncoding;
+        }
+
+        public java.util.Optional<String> getFileName() {
+            return java.util.Optional.ofNullable(fileName);
+        }
+
+        public String getMediaType() {
+            return mediaType;
+        }
+
+        public java.util.Optional<Source> getSource() {
+            return java.util.Optional.ofNullable(source);
+        }
+
+        public java.util.Optional<String> getTestCaseStartedId() {
+            return java.util.Optional.ofNullable(testCaseStartedId);
+        }
+
+        public java.util.Optional<String> getTestStepId() {
+            return java.util.Optional.ofNullable(testStepId);
+        }
+
+        public java.util.Optional<String> getUrl() {
+            return java.util.Optional.ofNullable(url);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Attachment that = (Attachment) o;
+            return 
+                body.equals(that.body) &&             
+                contentEncoding.equals(that.contentEncoding) &&             
+                java.util.Objects.equals(fileName, that.fileName) &&             
+                mediaType.equals(that.mediaType) &&             
+                java.util.Objects.equals(source, that.source) &&             
+                java.util.Objects.equals(testCaseStartedId, that.testCaseStartedId) &&             
+                java.util.Objects.equals(testStepId, that.testStepId) &&             
+                java.util.Objects.equals(url, that.url);            
+        }
+
+        @Override
+        public int hashCode() {
+            return java.util.Objects.hash(
+                body,
+                contentEncoding,
+                fileName,
+                mediaType,
+                source,
+                testCaseStartedId,
+                testStepId,
+                url
+            );
+        }
+
+        @Override
+        public String toString() {
+            return "Attachment{" +
+                "body=" + body +
+                ", contentEncoding=" + contentEncoding +
+                ", fileName=" + fileName +
+                ", mediaType=" + mediaType +
+                ", source=" + source +
+                ", testCaseStartedId=" + testCaseStartedId +
+                ", testStepId=" + testStepId +
+                ", url=" + url +
+                '}';
+        }
+    }
+
+    public static final class Duration {
+        private final Long seconds;
+        private final Long nanos;
+
+        @JsonCreator
+        public Duration(
+            Long seconds,
+            Long nanos
+        ) {
```

Comment:
At a glance this meets the requirements on:

https://github.com/FasterXML/jackson-modules-java8/tree/master/parameter-names#usage-example

Not sure why it would need the annotation

Reply:
That one doesn't need it, but the code generator adds the annotation to all the constructors.